h2o-py/h2o/estimators/estimator_base.py

```diff
@@ -280,6 +279,8 @@ def _make_parms(self, x=None, y=None, training_frame=None, offset_column=None, f
                         ignored_columns_set.add(ic)
             if x is None:
                 xset = set(names) - {y} - ignored_columns_set
+            elif hasattr(x, 'algo') and (x.algo=='infogram'):
```

Comment:
please, we don't want to see any `if (_.algo == 'xxx')` anymore in those abstract classes: each algo should be self-contained.
Here, this means adding a redirection:
```python
elif isinstance(x, ModelBase) and hasattr(self, '_extract_x_from_model'):
    xset = self._extract_x_from_model(x)
```

and in `gen_infogram.py` you should add a class extension:
```python
def _extract_x_from_model(self, model):
    features = model._model_json.get('output', {}).get('admissible_features')
    if features is None:
        raise ValueError("model %s doesn't have any admissible features" % model.key)
    return set(features)
```

Reply:
I copied your method.